src/pages/overlay/hooks/useScrolling.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+// Stop considering the user to be scrolling 100ms after the last scroll event
+const activityTimeout = 100;
+
+const useScrolling = () => {
+  // Allow subscriptions to scroll events
+  const [callbacks, setCallbacks] = useState<((e: Event) => void)[]>([]);
+  const on = useCallback((fn: (e: Event) => void) => {
+    setCallbacks((prev) => [...prev, fn]);
+  }, []);
+  const off = useCallback((fn: (e: Event) => void) => {
+    setCallbacks((prev) => prev.filter((f) => f !== fn));
+  }, []);
+
+  // Listen for the wheel event (user scrolling on desktop)
+  const onWheelTimer = useRef<NodeJS.Timeout | null>(null);
+  const onWheelActive = useRef(false);
+  const onWheel = useCallback(() => {
+    onWheelActive.current = true;
+
+    if (onWheelTimer.current) clearTimeout(onWheelTimer.current);
+    onWheelTimer.current = setTimeout(() => {
+      onWheelActive.current = false;
+    }, activityTimeout);
+  }, []);
+
+  // Listen for touch move events (user scrolling on mobile)
+  const onTouchMoveTimer = useRef<NodeJS.Timeout | null>(null);
+  const onTouchMoveActive = useRef(false);
+  const onTouchMove = useCallback(() => {
+    onTouchMoveActive.current = true;
+
+    if (onTouchMoveTimer.current) clearTimeout(onTouchMoveTimer.current);
+    onTouchMoveTimer.current = setTimeout(() => {
+      onTouchMoveActive.current = false;
+    }, activityTimeout);
+  }, []);
```

Comment:
Open to other suggestions on how to ensure the scroll event came from a user interaction.

This isn't perfect, keyboard navigation isn't captured here for example -- I suppose we could also listen for any key press and treat scrolls coming from that as from the user too?